### PR TITLE
update and delete sight and trip functionality

### PR DESCRIPTION
**Pull request by Shazeen15**

### CODE REVIEW

File: app/assets/stylesheets/footer.scss [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+        background-color:rgb(230, 190, 190);
+    }
+    .footer-link{
+        display:flex;
+        justify-content: flex-end;
+        color:rgb(251, 254, 251);
+        background-color: rgb(230, 190, 190) ;
+        text-decoration: none;
+    }
+}
```
Comment (minachuong) [inline app/assets/stylesheets/footer.scss:17]: please clean up tabbing and spacing 🧹

File: app/controllers/sights_controller.rb [hunk 2]
```diff
@@ -8,7 +8,27 @@ def create
             render json: sight.errors, status: :unprocessable_entity
         end
     end
+
+    def destroy
+        sight = Sight.find(params[:id])
+        sight.destroy
+        if sight.valid?
+        render json: sight
```
Comment (minachuong) [inline app/controllers/sights_controller.rb:16]: indentation ➡️

File: app/controllers/trips_controller.rb [hunk 3]
```diff
@@ -1,20 +1,67 @@
...
         trips = Trip.all
         render :json => trips.to_json(include: :sights)
     end
+    # def index 
+    #     trips = Trip.all
+    #     if user_signed_in? and current_user
+    #       render :json => trips.to_json(include: :sights)
+    #     end
+    #     render :json => trips.to_json(include: :sights)
+    # end
```
Comment (minachuong) [inline app/controllers/trips_controller.rb:21]: is this commented out momentarily?

File: app/controllers/trips_controller.rb [hunk 4]
```diff
@@ -1,20 +1,67 @@
...
+    def destroy
+      trip = Trip.find(params[:id])
+      sights = Sight.where({trip_id: trip.id})
+      
+      sights.each do |sight|
+        sight.destroy
+      end
+      
+      trip.destroy
+        if trip.valid?
```
Comment (minachuong) [inline app/controllers/trips_controller.rb:41]: indentation ↩️

File: app/javascript/components/pages/SightEdit.js [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+      form: {
+        name: this.props.sight.name,
+        address1: this.props.sight.address1,
+        city: this.props.sight.city,
+        zip_code: this.props.sight.zip_code,
+        state: this.props.sight.state,
+        country: this.props.sight.country,
+        phone: this.props.sight.phone,
+        trip_id: this.props.trip && this.props.trip.id,
+      },
```
Comment (minachuong) [inline app/javascript/components/pages/SightEdit.js:18]: when the keys match properties, you can destructure the whole sight object from props

File: app/javascript/components/pages/TripEdit.js [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+export class TripEdit extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      form: {
+        trip_name: this.props.trip.trip_name,
+        trip_location: this.props.trip.trip_location,
+        trip_date_range: this.props.trip.trip_date_range,
+        is_public: this.props.trip.is_public,
```
Comment (minachuong) [inline app/javascript/components/pages/TripEdit.js:13]: try to destructure the whole trip object from props and store to form

**Review by minachuong — APPROVED** [review]
please try to make smaller PRs. This PR contained three features which can be broken up into three PRs

File: app/controllers/trips_controller.rb (hunk 3 above)
Comment (Shazeen15) [reply app/controllers/trips_controller.rb:21]: yes it is. working on the logic still

File: app/controllers/sights_controller.rb (hunk 2 above)
Comment (Shazeen15) [reply app/controllers/sights_controller.rb:16]: okay will fix themmm

**Review by sjproctor — APPROVED** [review]

**Pull request merged**